### DISCUSSION TITLE
ProjectReunion and Bringing In Capabilities That Were Available on Windows IoT

### DISCUSSION THREAD

**Discussion by myokeeh** [microsoft/WindowsAppSDK#709] [Q&A]
Will ProjectReunion help me use GPIO, I2C, SPI "low level" devices on Windows 10 that were available on Windows IoT?

The ability that would like to have is to add GPIO, I2C, SPI devices on a standard Windows 10 laptop/PC. Think about being able to take button input (via events) or add a sensor to be used with an app.

The folks at maintaining `System.Devices.Gpio` (.NET IoT) pointed me towards buying a FT4222 dev board (UMFT4222EV is what I got), and was able to get it to work on a .NET 5 sample app and encouraged by that. I currently develop UWP apps but hit some roadblocks unable to reference `Iot.Device.Bindings` and getting it to work.

Will ProjectReunion allow my UWP get this capability with .NET 5/6, maintain the same look/feel AND be able to distribute via the Store?

Attached chat log with .NET IoT folks: [Chat.docx](https://github.com/microsoft/ProjectReunion/files/6274925/Chat.docx)

**Comment by jonwis** [comment]
Can you reattach as plain text, or just dump in a comment here?

I'll ping our devices folks and see what they say.

**Reply by myokeeh** [reply]
Plain text attached: [Chat.txt](https://github.com/microsoft/ProjectReunion/files/6279946/Chat.txt)

**Comment by driver1998** [comment]
Actually Windows.Devices.GPIO/I2C/SPI has been working on Desktop for quite a while, that's what we use in the Windows on Raspberry Pi project.

![微信图片_20210428171221](https://user-images.githubusercontent.com/22699485/116379896-0a5d5d00-a846-11eb-99d8-302eaf28c7e5.jpg)
You can get the test programs used here on https://github.com/driver1998/rpi3win10demos, designed to run on RPi3/RPi4.

The issue is it requires a table from **system firmware**, specifically a definition of RHProxy device in ACPI table, describing all the GPIO pins and I2C/SPI/UART controllers on the device. And this kind of description does not work very well on something like USB GPIO dongle or something.

https://docs.microsoft.com/en-us/windows/uwp/devices-sensors/enable-usermode-access

**Reply by myokeeh** [reply]
This is something I was also made aware of some time ago, but didn't delve deeper into yet. Thanks for summarizing for me. This helps. There are currently some issues with WoR that still needs to be addressed. For me one is not being able to reliably update to latest feature updates via WU.
